src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(itemID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Wrap(ctx, err, "deleting mail message")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// replies
+// ---------------------------------------------------------------------------
+
+// GetReplies retrieves a Messageable item.
+func (c Channels) GetReplies(
+	ctx context.Context,
+	teamID, channelID, itemID string,
+) (serialization.Parsable, error) {
+	replies, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(channelID).
+		Messages().
+		ByChatMessageId(itemID).
+		Replies().
+		Get(ctx, nil)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return replies, nil
+}
+
+func (c Channels) PostReply(
+	ctx context.Context,
+	teamID, containerID, messageID string,
+	body models.ChatMessageable,
+) (models.ChatMessageable, error) {
+	itm, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(messageID).
+		Replies().
+		Post(ctx, body, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating reply message")
+	}
+
+	if itm == nil {
+		return nil, clues.New("nil response reply to message creation").WithClues(ctx)
+	}
+
+	return itm, nil
+}
+
+func (c Channels) DeleteReply(
+	ctx context.Context,
+	teamID, itemID, containerID, replyID string,
+) error {
+	// deletes require unique http clients
+	// https://github.com/alcionai/corso/issues/2707
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(itemID).
+		Replies().
+		ByChatMessageId1(replyID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Wrap(ctx, err, "deleting mail message")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// Helpers
+// ---------------------------------------------------------------------------
+
+func MessageInfo(msg models.ChatMessageable, size int64) *details.GroupsInfo {
+	var (
+		created = ptr.Val(msg.GetCreatedDateTime())
+	)
+
+	return &details.GroupsInfo{
+		ItemType: details.GroupChannel,
+		Size:     size,
+		Created:  created,
+		Modified: ptr.OrNow(msg.GetLastModifiedDateTime()),
+	}
+}
+
+// ---------------------------------------------------------------------------
+// helper funcs
+// ---------------------------------------------------------------------------
+
+// ChannelsDisplayable is a wrapper that complies with the
+// models.Channelable interface with the graph.Container
+// interfaces.
+type ChannelsDisplayable struct {
```

Comment:
Something of a conflict of interests here.  Displayable is a concept we use for containers, not for items or drives.  This is related to the other comment checking whether we want to treat the container as an item or as a drive.  If we're going to match it to a displayable, it sounds like we want it to be a `Container`.  If we want it to be an Item, or similar to a drive, I don't think we want this compliance.

And, for that matter, I'm not sure this interface is valuable outside of the Exchange collection processor anyway.  I'd say see if we can cut it out, and only bring it back if necessary.